pincer/client.py

```diff
@@ -857,5 +860,139 @@ async def get_webhook(
+        """|coro|
+        The user object of the requester's account.
+
+        For OAuth2, this requires the ``identify`` scope,
+        which will return the object *without* an email,
+        and optionally the ``email`` scope,
+        which returns the object *with* an email.
+
+        Returns
+        -------
+        :class:`~pincer.objects.user.user.User`
+        """
+        return User.from_dict(
+            construct_client_dict(self, self.http.get("users/@me"))
+        )
+
+    async def modify_current_user(
+        self, username: Optional[str] = None, avatar: Optional = None
+    ) -> User:
+        """|coro|
+        Modify the requester's user account settings
+
+        Parameters
+        ----------
+        username : Optional[:class:`str`]
+            user's username,
+            if changed may cause the user's discriminator to be randomized.
+            |default| :data:`None`
+        avatar : Optional[:class:``]
+            if passed, modifies the user's avatar
+            a data URI scheme of JPG, GIF or PNG
+            |default| :data:`None`
+
+        Returns
+        -------
+        :class:`~pincer.objects.user.user.User`
+            Current modified user
+        """
+
+        user = await self.http.patch(
+            "users/@me", remove_none({"username": username, "avatar": avatar})
+        )
+        return User.from_dict(construct_client_dict(self, user))
+
+    async def get_current_user_guilds(
+        self,
+        before: Optional[Snowflake] = None,
+        after: Optional[Snowflake] = None,
+        limit: Optional[int] = None,
+    ) -> AsyncIterator[Guild]:
+        """|coro|
+        Returns a list of partial guild objects the current user is a member of.
+        Requires the ``guilds`` OAuth2 scope.
+
+        Parameters
+        ----------
+        before : Optional[:class:`~pincer.utils.snowflake.Snowflake`]
+            get guilds before this guild ID
+        after : Optional[:class:`~pincer.utils.snowflake.Snowflake`]
+            get guilds after this guild ID
+        limit : Optional[:class:`int`]
+                max number of guilds to return (1-200) |default| :data:`200`
+
+        Yields
+        ------
+        :class:`~pincer.objects.guild.guild.Guild`
+            A Partial Guild that the user is in
+
+        """
+        guilds = await self.http.get(
+            "users/@me/guilds?"
+            + (f"{before=}&" if before else "")
+            + (f"{after=}&" if after else "")
+            + (f"{limit=}&" if limit else "")
+        )
+
+        for guild in guilds:
+            yield Guild.from_dict(construct_client_dict(self, guild))
+
+    async def leave_guild(self, _id: Snowflake):
+        """|coro|
+        Leave a guild.
+
+        Parameters
+        ----------
+        _id : :class:`~pincer.utils.snowflake.Snowflake`
+            the id of the guild that the bot will leave
+        """
+        self.http.delete(f"users/@me/guilds/{_id}")
+
```

Comment:
```suggestion
    async def leave_guild(self, _id: Snowflake):
        """|coro|
        Leave a guild.

        Parameters
        ----------
        _id : :class:`~pincer.utils.snowflake.Snowflake`
            the id of the guild that the bot will leave
        """
        self.http.delete(f"users/@me/guilds/{_id}")
        self._client.guilds.pop(_id) 
```
I think some methods in the future will use `Client.guilds` to remove an API call but the info will be out of date since the client will stop receiving presence updates. Easiest solution would be removing the guild completely.